bot/glossary_command.go

```diff
@@ -0,0 +1,157 @@
+package bot
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/quintilesims/slackbot/db"
+	"github.com/quintilesims/slackbot/models"
+	glob "github.com/ryanuber/go-glob"
+	"github.com/urfave/cli"
+)
+
+// NewGlossaryCommand returns a cli.Command that manages !glossary
+func NewGlossaryCommand(store db.Store, w io.Writer) cli.Command {
+	return cli.Command{
+		Name:  "!glossary",
+		Usage: "manage the glossary",
+		Subcommands: []cli.Command{
+			{
+				Name:      "add",
+				Aliases:   []string{"a"},
+				Usage:     "add or set an entry in the glossary",
+				ArgsUsage: "KEY DEFINITION",
+				Action:    newGlossaryAddAction(store, w),
+			},
+			{
+				Name:      "remove",
+				Aliases:   []string{"rm"},
+				Usage:     "remove an entry from the glossary",
+				ArgsUsage: "KEY",
+				Action:    newGlossaryRemoveAction(store, w),
+			},
+			{
+				Name:      "search",
```

Comment:
Hmmm, so this is close to `ls` we've seen in other commands, like !alias ls. Do you think we should standardize on that? Like most commands have:
```
!command add [--force]
!command rm KEY
!command ls [--count] [GLOB]
```

Maybe I'm trying to shoehorn `search` into `ls` ?

Reply:
I think ls makes more sense, the language seems off right now. Either rm should be remove or search should change to ls